src/pages/ChatPage.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import './ChatPage.css';
+
+const ChatPage = ({location, history}) => {
+    const name = location.state;
+    const [message, setMessage] = useState([]);
+    const userText=['안녕하세요', '샘플 데이터입니다', '반가워요', '날씨가 어때요?', '감사합니다', '좋은하루 되세요', '사랑해요ㅎ.ㅎ'];
+    const [value, setValue] = useState('');
+    const onChange = (event) => {
+        setValue(event.target.value);
+    };
+    
+    const sendFunc = () => {
+        if(value === '')  return;
+        let tok = false;
+        let idx = 0;
+        if(Math.floor(Math.random() * 2)){
+            tok = true;
+            idx = Math.floor(Math.random() * userText.length);
+        }
+        if(tok){
+            setMessage([...message, {'isMine': true, 'text': value}, {'isMine': false, 'text': userText[idx]}]);
+        }else{
+            setMessage([...message, {'isMine': true, 'text': value}]);
+        }
+        setValue('');
+    }
+
+    return (
+        <div className="container">
+            <div className="header">
+                {name}
+            </div>
+            <div className="chatBox">
+                {message.map(item => {
+                    return item['isMine'] ? <div style={{textAlign: 'right'}}><div className="mySendBox">{item['text']}</div></div> : <div className="userSendBox">{item['text']}</div>
+                })}
```

Comment:
```suggestion
                {message.map(({isMine, text})=> {
                  return <div className={`${isMine === true 
                  ? styles.mySendBox  
                  : styles.yourSendBox}`}>{text}</div> })}
```
1. 삼항 연산자는 가장 작은 단위에서 써야 코드의 중복을 피할 수 있습니다.

2. map의 value 객체에 담긴 여러 필드에 참조를 반복해야 한다면 분해할당을 이용해 보세요.

3. mySendBox의 textAlign을 조정하기 위해 새로운 div를 사용하셨습니다. 단순히 클래스네임을 바꾸는 것으로 해결할 수 있습니다.

Reply:
분해할당에 대해 찾아보고 알게 됐어요. 감사합니다:)

Reply:
딱 정해놓고 말씀드릴 수 없는 것은 방법이 여러가지가 있기 때문입니다. 또 컴포넌트 내부에 어떤 요소가 들어가는지에 따라 다르겠지요. 좌우를 조정할 때에 현수님께서 제시한 방법도 유효합니다. 그러나 차라리 말씀드리고 싶었던 바는 매우 유사한 역할을 하는 동일한 두 객체는 그 구조도 유사해야 한다는 것입니다. textAlign을 바꾸기 위해 div를 하나 더 쓴다면, 너비와 높이, 간격 혹은 정렬에서 문제가 발생할 수 있습니다. 여기서 문제는 종종 동일한 결과가 산출이 되지 않는 것과, 유지보수에 있어서 번거로워 진다는 것입니다.